src/client/hud.cpp

```diff
@@ -1007,97 +1007,107 @@ void drawItemStack(
 	// Render as mesh if animated or no inventory image
 	if ((enable_animations && rotation_kind < IT_ROT_NONE) || def.inventory_image.empty()) {
 		ItemMesh *imesh = client->idef()->getWieldMesh(def.name, client);
-		if (!imesh || !imesh->mesh)
-			return;
-		scene::IMesh *mesh = imesh->mesh;
-		driver->clearBuffers(video::ECBF_DEPTH);
-		s32 delta = 0;
-		if (rotation_kind < IT_ROT_NONE) {
-			MeshTimeInfo &ti = rotation_time_infos[rotation_kind];
-			if (mesh != ti.mesh && rotation_kind != IT_ROT_OTHER) {
-				ti.mesh = mesh;
-				ti.time = porting::getTimeMs();
-			} else {
-				delta = porting::getDeltaMs(ti.time, porting::getTimeMs()) % 100000;
+		if (!imesh || !imesh->mesh) {
+			video::ITexture *texture = client->tsrc()->getTexture("no_texture.png");
+			video::SColor color = video::SColor(255, 255, 255, 255);
+			const video::SColor colors[] = { color, color, color, color };
+			draw2DImageFilterScaled(driver, texture, rect,
+				core::rect<s32>({0, 0},
+				core::dimension2di(texture->getOriginalSize())),
+				clip, colors, true);
```

Comment:
uh what?
It would be much better to set `inventory_image` to "no_texture.png" if none is defined and no mesh exists
instead of duplicating this code and also having to touch all those lines